src/utils/validation.ts

```diff
@@ -42,21 +42,26 @@ export const validateInputs = (
     return { displayName, publishedName };
 };
 
-const hasPublishedId = (filePath: string): boolean => {
-    const fileContentsString = readFileSync(filePath).toString();
-    const fileContents = JSON.parse(fileContentsString);
-    return !!fileContents.id;
-};
-
 export const validateNotAlreadyPublishedOrExit = (): void => {
-    if (hasPublishedId(BLOCK_SETTINGS_FILE)) {
+    const fileContents = readBlockSettingsFile(BLOCK_SETTINGS_FILE);
+
+    if (!!fileContents.id && fileContents.isStaging) {
```

Comment:
Why? `&& fileContents.isStaging`

If a user publish, no matter what, they can't publish again. (Only check for id needed)